pkg/apis/core/v1alpha1/helper/helper.go

```diff
@@ -779,3 +780,43 @@ func IsAPIServerExposureManaged(obj metav1.Object) bool {
 
 	return false
 }
+
+// GetSecretBindingTypes returns the SecretBinding provider types.
+func GetSecretBindingTypes(secretBinding *gardencorev1alpha1.SecretBinding) []string {
+	return strings.Split(secretBinding.Provider.Type, ",")
+}
+
+// SecretBindingHasType checks if the given SecretBinding has the given provider type.
+func SecretBindingHasType(secretBinding *gardencorev1alpha1.SecretBinding, toFind string) bool {
+	if secretBinding == nil {
+		return false
+	}
+	if secretBinding.Provider == nil {
+		return false
+	}
+
+	types := GetSecretBindingTypes(secretBinding)
+	if len(types) == 0 {
+		return false
+	}
+
+	return utils.ValueExists(toFind, types)
+}
+
+// AddTypeToSecretBinding adds the given provider type to the SecretBinding.
+func AddTypeToSecretBinding(secretBinding *gardencorev1alpha1.SecretBinding, toAdd string) {
+	if secretBinding.Provider == nil {
+		secretBinding.Provider = &gardencorev1alpha1.SecretBindingProvider{
+			Type: toAdd,
+		}
+		return
+	}
+
+	types := GetSecretBindingTypes(secretBinding)
+	if !utils.ValueExists(toAdd, types) {
+		types = append(types, toAdd)
+	}
+
+	newType := strings.Join(types, ",")
+	secretBinding.Provider.Type = newType
```

Comment:
Can we do,
```
if !utils.ValueExists(toAdd, types) {
	secretBinding.Provider.Type += "," + toAdd
}
```

Reply:
Actually, I would prefer the current logic as it is, it's more explicit and less error prone. I would just suggest to eliminate the short lived `newType` variable, e.g. 
```suggestion
	if !utils.ValueExists(toAdd, types) {
		types = append(types, toAdd)
	}
	secretBinding.Provider.Type = strings.Join(types, ",")
```